src/Compilers/CSharp/Test/Syntax/Parsing/LambdaAttributeParsingTests.cs

```diff
@@ -3637,5 +3642,875 @@ void verify(string source, ParseOptions? parseOptions = null)
+                                            }
+                                        }
+                                    }
+                                }
+                                M(SyntaxKind.SemicolonToken);
+                            }
+                            N(SyntaxKind.ExpressionStatement);
+                            {
+                                N(SyntaxKind.SimpleAssignmentExpression);
+                                {
+                                    N(SyntaxKind.IdentifierName);
+                                    {
+                                        N(SyntaxKind.IdentifierToken, "x");
+                                    }
+                                    N(SyntaxKind.EqualsToken);
+                                    N(SyntaxKind.CollectionExpression);
+                                    {
+                                        N(SyntaxKind.OpenBracketToken);
+                                        N(SyntaxKind.ExpressionElement);
+                                        {
+                                            N(SyntaxKind.NumericLiteralExpression);
+                                            {
+                                                N(SyntaxKind.NumericLiteralToken, "0");
+                                            }
+                                        }
+                                        N(SyntaxKind.CloseBracketToken);
+                                    }
+                                }
+                                M(SyntaxKind.SemicolonToken);
+                            }
+                            N(SyntaxKind.Block);
+                            {
+                                N(SyntaxKind.OpenBraceToken);
+                                N(SyntaxKind.CloseBraceToken);
+                            }
+                            N(SyntaxKind.EmptyStatement);
+                            {
+                                N(SyntaxKind.SemicolonToken);
+                            }
+                            N(SyntaxKind.CloseBraceToken);
+                        }
+                    }
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/73789")]
+        public void ParseAttributeWithLambaPathological()
+        {
+            // Ensure we can parse this pathological lambda attribute code without getting into exponential time.
+            int n = 30;
+            var builder = new StringBuilder();
+            for (int i = 0; i < n; i++)
+                builder.Append("[A(");
+            for (int i = 0; i < n; i++)
+                builder.Append(")]");
+            builder.Append("class C { }");
+
+            var tree = ParseTree(builder.ToString(), CSharpParseOptions.Default);
```

Comment:
note: this test no longer validates https://github.com/dotnet/roslyn/pull/73788/files#diff-a655d393d9799eabace5e5109383232749be8052f1fd825fc52b9f170908e838R11891.  We can remove that codee and this test passes (though the original user report still hangs the compiler).